### ISSUE TITLE
salt-cloud can't create snapshots, because there is a bug in the Unicode name of the virtual machine

### ISSUE THREAD

**Issue by konstest** [saltstack/salt#44452]
### Description of Issue/Question
If there is a VMware vCenter virtual machine whose name contains Unicode characters (not ASCII), salt-cloud can't create snapshots of some virtual machines

### Setup
Example name of the virtual machine with the UNICODE symbols that i got from the python console:
```
root@devops-ci-master:/# python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import salt.cloud
>>> import pprint
>>> client = salt.cloud.CloudClient('/etc/salt/cloud')
>>> nodes = client.query()
>>> for node in nodes['vcenter']['vmware']:
...     pprint.pprint(nodes['vcenter']['vmware'][node]['id'])
...     print(nodes['vcenter']['vmware'][node]['id'])
... 
 ----- many virt machines names --------
...
u'Windows Server 2003 Ent x86 SP2 EN Oracle10R2 (\u0421) PNovikov 220.128'
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\u0421' in position 47: ordinal not in range(128)
>>> 
```

### Steps to Reproduce Issue
```
root@devops-ci-master:/# salt-cloud -l debug -a create_snapshot server-ci-high-01.build.domain.name snapshot_name="Before_git_installation" memdump=False
[DEBUG   ] Reading configuration from /etc/salt/cloud
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/master.d/file_roots.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/file_roots.conf
[DEBUG   ] Including configuration from '/etc/salt/master.d/gitfs.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/gitfs.conf
[DEBUG   ] Including configuration from '/etc/salt/master.d/reactor.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/reactor.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: devops-ci-master.build.domain.name
[DEBUG   ] Missing configuration file: /etc/salt/cloud.providers
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/digitalocean.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/digitalocean.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/ec2.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/ec2.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/gogrid.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/gogrid.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/hpcloud-openstack.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/hpcloud-openstack.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/ibmsce.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/ibmsce.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/joyent.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/joyent.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/linode.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/linode.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/parallels.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/parallels.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/proxmox.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/proxmox.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/rackspace-openstack.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/rackspace-openstack.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/saltify.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/saltify.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/vcenter.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/vcenter.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/vsphere.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/vsphere.conf
[DEBUG   ] Missing configuration file: /etc/salt/cloud.profiles
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/gitlab-ci.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/gitlab-ci.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/linux-ci.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/linux-ci.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/linux.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/linux.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/services.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/services.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/windows-ci.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/windows-ci.conf
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/windows.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/windows.conf
[DEBUG   ] Configuration file path: /etc/salt/cloud
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] salt-cloud starting
[DEBUG   ] Could not LazyLoad parallels.avail_sizes: 'parallels' __virtual__ returned False
[DEBUG   ] LazyLoaded parallels.avail_locations
[DEBUG   ] LazyLoaded proxmox.avail_sizes
[DEBUG   ] Could not LazyLoad saltify.destroy: 'saltify.destroy' is not available.
[DEBUG   ] Could not LazyLoad saltify.avail_sizes: 'saltify.avail_sizes' is not available.
[DEBUG   ] Could not LazyLoad saltify.avail_images: 'saltify.avail_images' is not available.
[DEBUG   ] Could not LazyLoad saltify.avail_locations: 'saltify.avail_locations' is not available.
[DEBUG   ] Could not LazyLoad rackspace.reboot: 'rackspace.reboot' is not available.
[DEBUG   ] LazyLoaded openstack.list_locations
[DEBUG   ] Could not LazyLoad rackspace.list_locations: 'rackspace.list_locations' is not available.
The following virtual machines are set to be actioned with "create_snapshot":
  server-ci-high-01.build.domain.name

Proceed? [N/y] y
... proceeding
[DEBUG   ] Could not LazyLoad vmware.optimize_providers: 'vmware.optimize_providers' is not available.
[DEBUG   ] The 'vmware' cloud driver is unable to be optimized.
[DEBUG   ] Could not LazyLoad parallels.avail_sizes: 'parallels' __virtual__ returned False
[DEBUG   ] LazyLoaded parallels.avail_locations
[DEBUG   ] LazyLoaded proxmox.avail_sizes
[DEBUG   ] Could not LazyLoad saltify.destroy: 'saltify.destroy' is not available.
[DEBUG   ] Could not LazyLoad saltify.avail_sizes: 'saltify.avail_sizes' is not available.
[DEBUG   ] Could not LazyLoad saltify.avail_images: 'saltify.avail_images' is not available.
[DEBUG   ] Could not LazyLoad saltify.avail_locations: 'saltify.avail_locations' is not available.
[DEBUG   ] Could not LazyLoad rackspace.reboot: 'rackspace.reboot' is not available.
[DEBUG   ] LazyLoaded openstack.list_locations
[DEBUG   ] Could not LazyLoad rackspace.list_locations: 'rackspace.list_locations' is not available.
[DEBUG   ] vm:win2008r2 210.197 in provider:vmware is not in name list:'set(['server-ci-high-01.build.domain.name'])'

.... many vms ....

[DEBUG   ] vm:Ubuntu16.04x64 in provider:vmware is not in name list:'set(['server-ci-high-01.build.domain.name'])'
[ERROR   ] There was an error actioning machines: 'ascii' codec can't encode character u'\u0421' in position 47: ordinal not in range(128)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/cli.py", line 252, in run
    ret = mapper.do_action(names, kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1506, in do_action
    'list:\'{2}\''.format(vm_name, driver, names)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u0421' in pos
```

### Versions Report
```
root@devops-ci-master:/# salt-cloud --versions-report
Salt Version:
            Salt: 2017.7.1
 
Dependency Versions:
 Apache Libcloud: 1.5.0
            cffi: 1.5.2
        cherrypy: 11.0.0
        dateutil: 2.4.2
       docker-py: Not Installed
           gitdb: Not Installed
       gitpython: Not Installed
           ioflo: Not Installed
          Jinja2: 2.8
         libgit2: 0.24.0
         libnacl: Not Installed
        M2Crypto: Not Installed
            Mako: Not Installed
    msgpack-pure: Not Installed
  msgpack-python: 0.4.6
    mysql-python: Not Installed
       pycparser: 2.14
        pycrypto: 2.6.1
    pycryptodome: Not Installed
          pygit2: 0.24.0
          Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
    python-gnupg: Not Installed
          PyYAML: 3.11
           PyZMQ: 15.2.0
            RAET: Not Installed
           smmap: Not Installed
         timelib: Not Installed
         Tornado: 4.2.1
             ZMQ: 4.1.4
 
System Versions:
            dist: Ubuntu 16.04 xenial
          locale: ANSI_X3.4-1968
         machine: x86_64
         release: 4.4.0-96-generic
          system: Linux
         version: Ubuntu 16.04 xenial
 
```

**Comment by Ch3LL** [comment]
@konstest thanks for the PR! We will close up this issue once the PR is merged thanks :)

**Comment by damon-atkins** [comment]
The reason for this issue is you have non-ascii chars in your vm names but  your language on your salt master is not Unicode/UTF-8
System Versions:
            dist: Ubuntu 16.04 xenial
          **locale: ANSI_X3.4-1968**
         machine: x86_64
         release: 4.4.0-96-generic
          system: Linux
         version: Ubuntu 16.04 xenial

**Comment by whytewolf** [comment]
exporting the LANG to utf will not fix the issue this is bringing up. 
I have a system that has UTF-8 as the default lang and am still getting this error. 

```
Salt Version:
            Salt: 2017.7.4

Dependency Versions:
 Apache Libcloud: 2.0.0rc2
            cffi: Not Installed
        cherrypy: Not Installed
        dateutil: Not Installed
       docker-py: Not Installed
           gitdb: Not Installed
       gitpython: Not Installed
           ioflo: Not Installed
          Jinja2: 2.7.2
         libgit2: Not Installed
         libnacl: Not Installed
        M2Crypto: Not Installed
            Mako: Not Installed
    msgpack-pure: Not Installed
  msgpack-python: 0.5.1
    mysql-python: Not Installed
       pycparser: Not Installed
        pycrypto: 2.6.1
    pycryptodome: Not Installed
          pygit2: Not Installed
          Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
    python-gnupg: Not Installed
          PyYAML: 3.11
           PyZMQ: 15.3.0
            RAET: Not Installed
           smmap: Not Installed
         timelib: Not Installed
         Tornado: 4.2.1
             ZMQ: 4.1.4

System Versions:
            dist: centos 7.4.1708 Core
          locale: UTF-8
         machine: x86_64
         release: 3.10.0-693.11.6.el7.x86_64
          system: Linux
         version: CentOS Linux 7.4.1708 Core
```

```
[DEBUG   ] Reading configuration from /etc/salt/cloud
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: master.localhost
[DEBUG   ] Missing configuration file: /etc/salt/cloud.providers
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/vmware.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/vmware.conf
[DEBUG   ] Missing configuration file: /etc/salt/cloud.profiles
[TRACE   ] The required configuration section, 'fluent_handler', was not found the in the configuration. Not loading the fluent logging handlers module.
[TRACE   ] None of the required configuration sections, 'logstash_udp_handler' and 'logstash_zmq_handler', were found in the configuration. Not loading the Logstash logging handlers module.
[DEBUG   ] Configuration file path: /etc/salt/cloud
[TRACE   ] Trying pysss.getgrouplist for 'root'
[TRACE   ] Trying generic group list for 'root'
[TRACE   ] Group list for user 'root': '[]'
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] salt-cloud starting
[TRACE   ] 'parallels.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] Error loading clouds.libvirt: Unable to locate or import python libvirt library.
[TRACE   ] Error loading clouds.virtualbox: The virtualbox driver cannot be loaded: 'vboxapi' is not installed.
[DEBUG   ] Could not LazyLoad parallels.avail_sizes: 'parallels' __virtual__ returned False
[TRACE   ] 'parallels.avail_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] LazyLoaded parallels.avail_locations
[TRACE   ] 'proxmox.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] LazyLoaded proxmox.avail_sizes
[TRACE   ] 'saltify.destroy' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.destroy: 'saltify.destroy' is not available.
[TRACE   ] 'saltify.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.avail_sizes: 'saltify.avail_sizes' is not available.
[TRACE   ] 'saltify.avail_images' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.avail_images: 'saltify.avail_images' is not available.
[TRACE   ] 'saltify.avail_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.avail_locations: 'saltify.avail_locations' is not available.
[TRACE   ] 'rackspace.reboot' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad rackspace.reboot: 'rackspace.reboot' is not available.
[TRACE   ] 'openstack.list_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] LazyLoaded openstack.list_locations
[TRACE   ] 'rackspace.list_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad rackspace.list_locations: 'rackspace.list_locations' is not available.
[DEBUG   ] Could not LazyLoad vmware.optimize_providers: 'vmware.optimize_providers' is not available.
[DEBUG   ] The 'vmware' cloud driver is unable to be optimized.
The following virtual machines are set to be actioned with "create_snapshot":
  test3

Proceed? [N/y] ... proceeding
[TRACE   ] 'parallels.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[TRACE   ] Error loading clouds.libvirt: Unable to locate or import python libvirt library.
[TRACE   ] Error loading clouds.virtualbox: The virtualbox driver cannot be loaded: 'vboxapi' is not installed.
[DEBUG   ] Could not LazyLoad parallels.avail_sizes: 'parallels' __virtual__ returned False
[TRACE   ] 'parallels.avail_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] LazyLoaded parallels.avail_locations
[TRACE   ] 'proxmox.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] LazyLoaded proxmox.avail_sizes
[TRACE   ] 'saltify.destroy' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.destroy: 'saltify.destroy' is not available.
[TRACE   ] 'saltify.avail_sizes' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.avail_sizes: 'saltify.avail_sizes' is not available.
[TRACE   ] 'saltify.avail_images' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.avail_images: 'saltify.avail_images' is not available.
[TRACE   ] 'saltify.avail_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad saltify.avail_locations: 'saltify.avail_locations' is not available.
[TRACE   ] 'rackspace.reboot' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad rackspace.reboot: 'rackspace.reboot' is not available.
[TRACE   ] 'openstack.list_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] LazyLoaded openstack.list_locations
[TRACE   ] 'rackspace.list_locations' has been marked as not supported. Removing from the list of supported cloud functions
[DEBUG   ] Could not LazyLoad rackspace.list_locations: 'rackspace.list_locations' is not available.
[TRACE   ] Retrieving new service instance
[TRACE   ] Connecting using the 'userpass' mechanism, with username 'fakeuser'
[TRACE   ] Checking connection is still authenticated
[TRACE   ] Retrieving root folder
[TRACE   ] Retrieved 7 objects
[DEBUG   ] vm:cloud-tests-template-suse in provider:vmware is not in name list:'set(['test3'])'
[DEBUG   ] vm:test2 in provider:vmware is not in name list:'set(['test3'])'
[ERROR   ] There was an error actioning machines: 'ascii' codec can't encode character u'\u2013' in position 4: ordinal not in range(128)
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/cloud/cli.py", line 253, in run
    ret = mapper.do_action(names, kwargs)
  File "/usr/lib/python2.7/site-packages/salt/cloud/__init__.py", line 1497, in do_action
    'list:\'{2}\''.format(vm_name, driver, names)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2013' in position 4: ordinal not in range(128)
```

in this test `test3` doesn't actually exist. it is just used to force the vmware driver to iterate through the instances. and there exists a server in the vmware setup that is named test–server with the unicode 2013 dash instead of a hyphen. 

Also as you can see the system lang is already UTF-8.

While testing this. I edited the line in question to drop the .format in favor of logs internal formatting. and the error went away. even when the default locale is  ANSI_X3.4-1968

**Comment by whytewolf** [comment]
ZD-2309

**Comment by rallytime** [comment]
Fixed with #46455
